bitcoin_attacker_&_2_honest_pools.py

```
@@ -240,215 +195,208 @@ def get_state_transitions(self, state: BlockchainModel.State, action: Blockchain
+                    taken_g = self.max_fork - vacant_g
+                    if att_down > blue and vacant_b > (blue + 1) and att_up > green and vacant_g > (green + 1):
+                        next_state = self.combine_state(att_up + 1, green, att_down, blue, v_ag, v_ab,
+                                                        self.Fork.Irrelevant,
+                                                        self.Fork.Active)
+                        transitions.add(next_state, probability=self.alpha)
+
+                        next_state = self.combine_state(att_up - green, 1, att_down, blue, self.set_block_status(v_ag, green, self.Miner.Attacker, taken_g), v_ab,
+                                                        self.Fork.Relevant, fork_blue)
+                        transitions.add(next_state, probability=(self.gamma * self.WW))
+
+                        next_state = self.combine_state(att_up, green + 1, att_down, blue, v_ag, v_ab, self.Fork.Relevant, fork_blue)
+                        transitions.add(next_state, probability=((1 - self.gamma) * self.WW))
+
+                        next_state = self.combine_state(att_up, green, att_down, blue + 1, v_ag, v_ab, fork_green, self.Fork.Relevant)
+                        transitions.add(next_state, probability=((1 - self.gamma) * (1 - self.WW - self.alpha)))
+
+                        next_state = self.combine_state(att_up, green, att_down - blue, 1, v_ag, self.set_block_status(v_ab, blue, self.Miner.Attacker, taken_b), fork_green, self.Fork.Relevant)
+                        transitions.add(next_state, probability=(self.gamma * (1 - self.WW - self.alpha)))
+
+                    else:
+                        transitions.add(self.final_state, probability=1, reward=self.error_penalty)
+                else:
+                    transitions.add(self.final_state, probability=1, reward=self.error_penalty)
+            else:
+                transitions.add(self.final_state, probability=1, reward=self.error_penalty)
```

This line should never be reached. Raise an error if you do reach it instead of returning a default illegal transition.